protocol/routing/discovery.md

```diff
@@ -0,0 +1,41 @@
+IPFS Peer Discovery Protocol Spec
+=================================
+
+Authors: David Dias
+
+Reviewers:
+
+TODOS:
+
+-----------------------
+
+> 
+
+## Supports
+
+- Peer discovery through
+  - mdns
+  - custom peers list
+  - random walking on the network
+
+## Overview
+
+
+### bootstrap peer list
+
+List with known and trusted peers shipped with IPFS.
+
+- _How is this list updated?_
+- _Is this list updated periodically_?
+
+### random walk
+
+IPFS issues random Peer lookups periodically to refresh our kbucket if needed. For impl reference, see: https://github.com/ipfs/go-ipfs/blob/master/routing/dht/dht_bootstrap.go#L88-L109.
```

Comment:
- we won't have a kbucket if it's not kademlia.
- random-walk is DHT-specific (or "query-thing-specific"), so make a note of that. maybe even call it "dht-random-walk"

> On nodes running a well-formed network with queries which expose more peers (e.g. DHT), IPFS can issue random queries to discover more nodes along those paths. This is useful for quickly filling a routing table with other nodes.